### DISCUSSION TITLE
Embed source text within output

### DISCUSSION THREAD

**Discussion by avidseeker** [jgm/pandoc#9915] [Ideas]
The final output whether it is pdf, html, or epub, is nice to view but often not very nice to edit and review.

Use case: I'd like to send a friend a nicely formatted document in pdf, and I want at the same time give him the freedom of extracting the source text that was used in making that pdf document.

html can include the source text by having it commented within itself. Epub is a zip so that's easy too. PDF is binary, but the source text can be appended to the end (see [SO972001](https://stackoverflow.com/questions/972001))

As for extracting the source text from the pdf, html, or epub, pandoc can make up a standard to locate where the source text would be. E.g:

* HTML look for this comment format:
```
<!--PANDOC_SRC
source code here
-->

the rest of HTML render
```
* EPUB: look for `PANDOC_SRC.md` file within the zip
* PDF: dump the appended text starting with PANDOC_SRC.

**Comment by TomBener** [comment]
I don't think your proposal is very good and it is even against the principle of these document formats.

Embedding source text in the generated documents is redundant. Rather, one can utilize the ability of Pandoc to generate multiple formats from the same source file, `.docx` file, for example, is generally used for editing. It is also possible to use the [--strip-comments](https://pandoc.org/MANUAL.html#option--strip-comments[) option to control comments in the html output. For PDF, you can [embed](https://help.adobe.com/en_US/framemaker/2020/using/using-framemaker/user-guide/frm_text_tx-attach-embed-files.html) the source text in PDF, even other output formats, so that you only need to share a PDF to others.
